### PR TITLE
feat(Condensed): the presheaf of locally constant maps as a condensed set

### PR DESCRIPTION
**Pull request by dagurtomas**
We define the functor `Condensed.LocallyConstant.functor : Type (u+1) ⥤ CondensedSet.{u}` which sends a set `X` to the sheaf of locally constant maps into `X`. 

Further, we prove that this functor is left adjoint to `Condensed.underlying`, and hence isomorphic to
`Condensed.discrete`. The unit of this adjunction is an isomorphism, which yields a proof that
`Condensed.discrete` is fully faithful.

---
- [x] depends on: #12155 
- [x] depends on: #12182 
- [x] depends on: #12186 
- [ ] depends on: #12870 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Condensed/Discrete/LocallyConstant.lean [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+  }
+  map f := ⟨functorToPresheaves.map f⟩
+
+/--
+`Condensed.LocallyConstant.functor` is naturally isomorphic to the restriction of
+`topCatToCondensed` to discrete topological spaces.
+-/
+noncomputable def functorIsoTopCatToCondensed : functor ≅ TopCat.discrete ⋙ topCatToCondensed :=
+  NatIso.ofComponents (fun X ↦ {
+    hom := ⟨(functorToPresheavesIsoTopCatToCondensed X).hom⟩
```
Comment (joelriou) [inline Mathlib/Condensed/Discrete/LocallyConstant.lean:122]: It should be possible to use `Functor.preimageIso` here (applied to the forget functor from sheaves to presheaves).

Comment (dagurtomas) [reply Mathlib/Condensed/Discrete/LocallyConstant.lean]: That works, thanks!

**Comment by dagurtomas** [comment]
Marked as WIP because I want to generalise a bit so that it more easily applies to light condensed sets as well

**Comment by dagurtomas** [comment]
Incorporated into #14027

**Pull request closed without merging**